beamer/chain.py

```diff
@@ -183,97 +184,85 @@ def _process_events(self) -> None:
             if not any_state_changed:
                 break
 
-    # TODO: pull this out of the event processor, so it can be tested independently
-    def _process_requests(self) -> None:
-        assert self._synced, "Not synced yet"
```

Comment:
This line is missing from the new function. But honestly I don't even get why we added it, if the caller ensures we are synced anyways. So was that the reason you removed it?